### ISSUE TITLE
[Bug]: PC0030 Field is flagged even though it is part of the primary key.

### ISSUE THREAD

**Issue by elweilando** [ALCops/Analyzers#456]
### Analyzer

PlatformCop

### Issue type

False positive (diagnostic fires when it shouldn't)

### Rule ID

PC0030

### Description

For the Sales Line, the “Line No.” field is flagged even though it is part of the primary key. According to the documentation, primary key fields do not need to be defined separately, as they are already included.

### AL code to reproduce

```al
    local procedure GetLastLineNo(SalesHeader: Record "Sales Header"): Integer
    var
        SalesLine: Record "Sales Line";
    begin
        SalesLine.SetRange("Document Type", SalesHeader."Document Type");
        SalesLine.SetRange("Document No.", SalesHeader."No.");
        if SalesLine.FindLast() then
            exit(SalesLine."Line No.")
        else
            exit(0);
    end;
```

### Additional context

<img width="1180" height="211" alt="Image" src="https://github.com/user-attachments/assets/96a9e7a8-ff4e-4830-9b1d-e7d388ea641a" />

**Comment by pri-kise** [comment]
If you don't specify SetLoadFields all Fields will be loaded.
Even if you only use the primary key fields.
In those special cases you need to use SetLoadFields on a primary key field to limit the loading of all fields.

**Comment by Arthurvdv** [comment]
What @pri-kise says here ☝️

If you use the Code Action `ALCops: Add SetLoadFields` it will resolve this for you and add the `SetLoadFields` for you.

```AL
procedure GetLastLineNo(SalesHeader: Record "Sales Header"): Integer
var
    SalesLine: Record "Sales Line";
begin
    SalesLine.SetRange("Document Type", SalesHeader."Document Type");
    SalesLine.SetRange("Document No.", SalesHeader."No.");
    SalesLine.SetLoadFields("Line No.");
    if SalesLine.FindLast() then
        exit(SalesLine."Line No.")
    else
        exit(0);
end;
```

**Comment by Arthurvdv** [comment]
@elweilando, did you happen to have a chance to take a look at the feedback here?

**Comment by Arthurvdv** [comment]
Closing this issue due to inactivity. If this is still relevant, feel free to reopen it and we can pick up the discussion again.
